MontyHallv2/DoorCreation/Door.cs

```diff
@@ -0,0 +1,42 @@
+namespace MontyHallv2.DoorCreation;
+
+public class Door
+{
+    private bool _opened;
+    private bool _hasCar;
+    private bool _playerPicked;
+
+    public void OpeningDoor()
+    {
+        _opened = true;
+    }
+
+    public bool HasPlayerPicked()
+    {
+        return _playerPicked;
+    }
+
+    public void PlayerPickedDoor()
+    {
+        _playerPicked = true;
+    }
+
+    public void PlayerUnpickDoor()
+    {
+        _playerPicked = false;
+    }
```

Comment:
if you're not keen on a big refactor as suggested with the comment on line 7, perhaps consider some renaming of these functions

```suggestion
    public bool IsPicked()
    {
        return _playerPicked;
    }

    public void PickDoor()
    {
        _playerPicked = true;
    }

    public void UnpickDoor()
    {
        _playerPicked = false;
    }
```